### PR TITLE
[HDInsight]-Add descriptions for sparkjob related definition.

### PR DESCRIPTION
**Pull request by sunsw1994**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Service team MUST add the "**WaitForARMFeedback**" label if the management plane API changes fall into one of the below categories. 
  - adding/removing APIs.
  - adding/removing properties.
  - adding/removing API-version. 
  - adding a new service in Azure.

<i>Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.</i>

- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/hdinsight/data-plane/Microsoft.HDInsight/preview/2018-11-01-preview/livySpark.json [hunk 1]
```diff
@@ -705,14 +705,17 @@
       "type": "object",
       "properties": {
         "from": {
+          "description": "The start index to fetch sessions.",
```
Comment (aim-for-better) [inline specification/hdinsight/data-plane/Microsoft.HDInsight/preview/2018-11-01-preview/livySpark.json:708]: >sessions [](start = 51, length = 8)

spark batch Job?

File: specification/hdinsight/data-plane/Microsoft.HDInsight/preview/2018-11-01-preview/livySpark.json [hunk 2]
```diff
@@ -705,14 +705,17 @@
...
           "format": "int32",
           "type": "integer"
         },
         "total": {
+          "description": "Number of sessions to fetch.",
           "format": "int32",
           "type": "integer"
         },
         "sessions": {
+          "description": "Batch list.",
```
Comment (aim-for-better) [inline specification/hdinsight/data-plane/Microsoft.HDInsight/preview/2018-11-01-preview/livySpark.json:718]: >Batch list [](start = 26, length = 10)

The Spark Batch job list.

File: specification/hdinsight/data-plane/Microsoft.HDInsight/preview/2018-11-01-preview/livySpark.json [hunk 3]
```diff
@@ -725,22 +728,27 @@
       "type": "object",
       "properties": {
         "id": {
+          "description": "The session id.",
```
Comment (aim-for-better) [inline specification/hdinsight/data-plane/Microsoft.HDInsight/preview/2018-11-01-preview/livySpark.json:731]: >session [](start = 30, length = 7)

The spark job's livy id.

File: specification/hdinsight/data-plane/Microsoft.HDInsight/preview/2018-11-01-preview/livySpark.json [hunk 4]
```diff
@@ -725,22 +728,27 @@
...
           "format": "int32",
           "type": "integer"
         },
         "appId": {
+          "description": "The application id of this session.",
```
Comment (aim-for-better) [inline specification/hdinsight/data-plane/Microsoft.HDInsight/preview/2018-11-01-preview/livySpark.json:736]: >session [](start = 53, length = 7)

spark job

File: specification/hdinsight/data-plane/Microsoft.HDInsight/preview/2018-11-01-preview/livySpark.json [hunk 5]
```diff
@@ -725,22 +728,27 @@
...
         "appInfo": {
+          "description": "The detailed application info.",
           "type": "object",
           "additionalProperties": {
             "type": "string"
           }
         },
         "state": {
-          "type": "string"
+          "description": "The batch state.",
```
Comment (aim-for-better) [inline specification/hdinsight/data-plane/Microsoft.HDInsight/preview/2018-11-01-preview/livySpark.json:747]: >The batch state [](start = 26, length = 15)

The spark batch job state.

File: specification/hdinsight/data-plane/Microsoft.HDInsight/preview/2018-11-01-preview/livySpark.json [hunk 6]
```diff
@@ -725,22 +728,27 @@
...
+          "$ref": "#/definitions/SessionState"
```
Comment (aim-for-better) [inline specification/hdinsight/data-plane/Microsoft.HDInsight/preview/2018-11-01-preview/livySpark.json:748]: >SessionState [](start = 33, length = 12)

I noticed that spark batch job and spark session job share the same SessionState. Could you please double check this?
If they don't have the totally same state please split it.

File: specification/hdinsight/data-plane/Microsoft.HDInsight/preview/2018-11-01-preview/livySpark.json [hunk 7]
```diff
@@ -786,45 +800,55 @@
           }
         },
         "files": {
+          "description": "Files to be used in this session.",
```
Comment (aim-for-better) [inline specification/hdinsight/data-plane/Microsoft.HDInsight/preview/2018-11-01-preview/livySpark.json:803]: >session [](start = 51, length = 7)

I know that the description in livy doc used the word "session". But this property is one of SparkBatchJobRequest. So I think that we need to mitigate "session" to "spark batch job". How do you like?

File: specification/hdinsight/data-plane/Microsoft.HDInsight/preview/2018-11-01-preview/livySpark.json (hunk 1 above)
Comment (sunsw1994) [reply specification/hdinsight/data-plane/Microsoft.HDInsight/preview/2018-11-01-preview/livySpark.json:708]: I add description following [livy documatation](https://livy.apache.org/docs/latest/rest-api.html)(search GET /batches in page)

Is it ok for different from this doc?

File: specification/hdinsight/data-plane/Microsoft.HDInsight/preview/2018-11-01-preview/livySpark.json [hunk 8]
```diff
@@ -895,29 +930,23 @@
...
-            "name": "SessionJobKind",
-            "modelAsString": true
-          },
-          "enum": [
-            "spark",
-            "pyspark",
-            "sparkr",
-            "sql"
-          ]
+          "description": "Session kind.",
```
Comment (aim-for-better) [inline specification/hdinsight/data-plane/Microsoft.HDInsight/preview/2018-11-01-preview/livySpark.json:933]: >Session [](start = 26, length = 7)

Spark Session kind?

File: specification/hdinsight/data-plane/Microsoft.HDInsight/preview/2018-11-01-preview/livySpark.json [hunk 9]
```diff
@@ -1010,18 +1054,21 @@
       "type": "object",
       "properties": {
         "id": {
+          "description": "Session id.",
```
Comment (aim-for-better) [inline specification/hdinsight/data-plane/Microsoft.HDInsight/preview/2018-11-01-preview/livySpark.json:1057]: >Session [](start = 26, length = 7)

The "Session" is so common. I think that we had better to add prefix for example Spark Batch, Spark Session, Spark Statement etc. 
Could you please check all the "session" word to make it clear enough?

File: specification/hdinsight/data-plane/Microsoft.HDInsight/preview/2018-11-01-preview/livySpark.json (hunk 1 above)
Comment (aim-for-better) [reply specification/hdinsight/data-plane/Microsoft.HDInsight/preview/2018-11-01-preview/livySpark.json:708]: I think that the minor change is acceptable and necessary to fit in our sdk. Although the class name is SparkBatchJobCollection.

---
In reply to: [455539008](https://github.com/Azure/azure-rest-api-specs/pull/10121#discussion_r455539008) [](ancestors = 455539008)

File: specification/hdinsight/data-plane/Microsoft.HDInsight/preview/2018-11-01-preview/livySpark.json (hunk 2 above)
Comment (sunsw1994) [reply specification/hdinsight/data-plane/Microsoft.HDInsight/preview/2018-11-01-preview/livySpark.json:718]: Thanks for your reminder. I will scan all description again to enhance the completeness.

File: specification/hdinsight/data-plane/Microsoft.HDInsight/preview/2018-11-01-preview/livySpark.json [hunk 10]
```diff
@@ -705,14 +705,17 @@
...
           "format": "int32",
           "type": "integer"
         },
         "total": {
+          "description": "Number of Spark Batch jobs to fetch.",
           "format": "int32",
           "type": "integer"
         },
         "sessions": {
+          "description": "The Spark Batch job list.",
```
Comment (aim-for-better) [inline specification/hdinsight/data-plane/Microsoft.HDInsight/preview/2018-11-01-preview/livySpark.json:718]: >job [](start = 42, length = 3)

jobs

File: specification/hdinsight/data-plane/Microsoft.HDInsight/preview/2018-11-01-preview/livySpark.json (hunk 3 above)
Comment (aim-for-better) [reply specification/hdinsight/data-plane/Microsoft.HDInsight/preview/2018-11-01-preview/livySpark.json]: I think livy is necessary to make it clear.
Do you think "The livy id of the spark batch job." will be better?

---
In reply to: [455536463](https://github.com/Azure/azure-rest-api-specs/pull/10121#discussion_r455536463) [](ancestors = 455536463)

File: specification/hdinsight/data-plane/Microsoft.HDInsight/preview/2018-11-01-preview/livySpark.json (hunk 10 above)
Comment (sunsw1994) [reply specification/hdinsight/data-plane/Microsoft.HDInsight/preview/2018-11-01-preview/livySpark.json]: Updated.

**Comment by aim-for-better** [comment]
Hi @anuchandy  Could you please review the PR?

**Review by aim-for-better — APPROVED** [review]
LGTM

**Review by majastrz — APPROVED** [review]
Changes is just backfilling missing swagger (changing strings to enums to match real API). Signed off from ARM side.

**Review by anuchandy — APPROVED** [review]

**Pull request merged**